pkg/apis/controller/common/v1alpha3/common_types.go

```diff
@@ -65,3 +69,57 @@ type Observation struct {
 	// Key-value pairs for metric names and values
 	Metrics []Metric `json:"metrics"`
 }
+
+// +k8s:deepcopy-gen=true
+type SourceSpec struct {
+	// Model-train source code can expose metrics by http, such as HTTP endpoint in
+	// prometheus metric format
+	HttpGet *v1.HTTPGetAction `json:"httpGet,omitempty"`
+	// During training model, metrics may be persisted into local file in source
+	// code, such as tfEvent use case
+	FileSystemPath *FileSystemPath `json:"fileSystemPath,omitempty"`
+	// Default metric output format is {"metric": "<metric_name>",
+	// "value": <int_or_float>, "epoch": <int>, "step": <int>}, but if the output doesn't
+	// follow default format, please extend it here
+	Filter *FilterSpec `json:"filter,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type FilterSpec struct {
+	// When the metrics output follows format as this field specified, metricsCollector
+	// collects it and reports to metrics server, it can be "<metric_name>: <float>" or else
+	MetricsFormat []string `json:"metricsFormat,omitempty"`
+}
+
+type FileSystemKind string
+
+const (
+	DirectoryKind FileSystemKind = "diretory"
```

Comment:
Typo: directory

Reply:
get it, I will include it in a coming PR soon.